pkg/codegen/utilities.go

```diff
@@ -79,3 +82,24 @@ func SortedKeys(m interface{}) []string {
 
 	return keys
 }
+
+// CleanDir removes all existing files from a directory except those in the exclusions list.
```

Comment:
One thing that might warrant a note is that the exclusions don't function recursively. For example, you cannot exclude a single file in a subdirectory--instead, you need to exclude the whole subdirectory.

On that note, is that the behavior we want, or do we need to be able to exclude specific files in subdirectories without excluding the whole directory?

Reply:
This is true but it hasn't come up yet. In fact, on the tfgen side of this we actually don't even need to add the overlays to the exclusions set because they are read into memory before the directory is cleaned. 

We can solve for that case within this function if we need to exclude files within subdirectories down the line but for now this is okay? Do you want me to add a comment to clarify any of this?

Reply:
Let's add a comment for clarification and leave the implementation as-is 👍